bindings/pyroot/cppyy/cppyy-backend/clingwrapper/src/clingwrapper.cxx

```diff
@@ -854,9 +856,12 @@ char* Cppyy::CallS(
         cstr = cppstring_to_cstring(*cppresult);
         *length = cppresult->size();
         cppresult->std::string::~basic_string();
-    } else
-        *length = 0;
-    free((void*)cppresult);
+        free((void *)cppresult);
+    } else {
+       *length = 0;
+       free((void *)cppresult);
```

Comment:
Seeing this, and although the code was there before...  Given that `free()` is called on `cppresult` in any case, can't we use instead a `std::unique_ptr` that ensures memory is released even when `throw`ing?